File: 2023/java/src/test/java/org/frieder/aoc/day1/TestDay1A.java
package org.frieder.aoc.day1;

import org.frieder.aoc.day1.a.Solution1A;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertEquals;

public class TestDay1A
{
    @Test
    public void testExample() throws IOException {
        assertEquals(142, Solution1A.getResult("../resources/1/test_a.txt"));
    }

    @Test
    public void testResult() throws IOException {
        assertEquals(56506, Solution1A.getResult("../resources/1/input.txt"));
    }
}

File: 2023/php/test/TestDay1A.php
<?php declare(strict_types=1);

$loader = require __DIR__ . '/../vendor/autoload.php';

use day1\a\Solution1A;
use PHPUnit\Framework\TestCase;
use utils\TestLogger;

final class TestDay1A extends TestCase
{

    public function testExample(): void
    {
        $this->assertSame(142, Solution1A::getResult("../resources/1/test_a.txt", TestLogger::createTestLogger()));
    }

    public function testResult(): void
    {
        $this->assertSame(56506, Solution1A::getResult("../resources/1/input.txt", TestLogger::createTestLogger()));
    }

}